File: src/Core/CardData.ts
import { CardSet, CardRarity } from "./Enums";
import type { ActionCommand, GridState, FortunaliaEngine } from "./FortunaliaEngine";

export interface ICardEffect {
    execute(command: ActionCommand, gridState: GridState, engine: FortunaliaEngine): void;
}

export interface CardData {
    id: string;
    name: string;
    rarity: CardRarity;
    icon: string;
    cardSet: CardSet;
    baseWeight: number;
    baseDamage: number;
    isInitiator: boolean;
    isFinisher: boolean;
    logic?: ICardEffect;
}

File: FortunaliaUnityMVP/Data/CardData.cs
using UnityEngine;

namespace Fortunalia.Data
{
    [CreateAssetMenu(fileName = "NewCard", menuName = "Fortunalia/Card Data")]
    public class CardData : ScriptableObject
    {
        public string id;
        public CardSet cardSet;
        
        [Tooltip("Базовый вес карты для расчета вероятности выпадения")]
        public float baseWeight = 1.0f;
        
        [Tooltip("Базовый наносимый урон кешу босса")]
        public float baseDamage = 10f;
        
        [Header("Behavior Flags")]
        [Tooltip("Выполняется самым первым (в начале стека)")]
        public bool isInitiator;
        [Tooltip("Выполняется самым последним (в конце стека), бонус от пройденных фаз")]
        public bool isFinisher;
        
        [SerializeReference]
        [Tooltip("Реализация логики карты")]
        public ICardEffect logic;
    }
}
